env/build_test.go

```diff
@@ -42,89 +49,120 @@ func testBuildEnv(t *testing.T, when spec.G, it spec.S) {
 	})
 
 	when("#NewBuildEnv", func() {
-		when("platform is latest", func() {
-			it.Before(func() {
-				platform.EXPECT().SupportsAssetPackages().Return(true)
-			})
-			it("includes expected vars", func() {
-				benv := env.NewBuildEnv([]string{
-					"CNB_STACK_ID=some-stack-id",
-					"HOSTNAME=some-hostname",
-					"HOME=some-home",
-					"HTTPS_PROXY=some-https-proxy",
-					"https_proxy=some-https-proxy",
-					"HTTP_PROXY=some-http-proxy",
+		it("includes expected vars", func() {
```

Comment:
I changed the nesting of tests in order to not repeat this assertion a bunch of times for the various combos of platform & buildpack. It's probably easiest to view this diff with main.